go-controller/pkg/clustermanager/network_cluster_controller.go

```diff
@@ -571,7 +571,11 @@ func (h *networkClusterControllerEventHandler) UpdateResource(oldObj, newObj int
 			return fmt.Errorf("could not cast %T object to *corev1.Node", newObj)
 		}
 		_, nodeFailed := h.nodeSyncFailed.Load(newNode.GetName())
-		if !nodeFailed && util.NoHostSubnet(oldNode) != util.NoHostSubnet(newNode) {
+		// Note: (trozet) It might be pedantic to check if the NeedsNodeAllocation. This assumes one of the following:
+		// 1. we missed an add event (bug in kapi informer code)
+		// 2. a user removed the annotation on the node
```

Comment:
current checks will only work if the annotation was removed, but not if it is incorrect (can happen on both missed kapi event after a restart and user changing annotation instead of deleting it).
not sure how much of a problem it solves for the price

Reply:
yeah I was a bit torn on this one, so I tried to compromise with a somewhat fast check and only solving if the annotations were removed. We don't support anything if a user changes the ovnk annotations. I guess we should also not support if they are removed, but I'm worried something may happen where we are essentially "stuck" and the node has no annotation, so if you are OK with it I'd prefer to have this code at least for now